corehq/apps/domain/forms.py

```diff
@@ -2377,3 +2383,91 @@ def __init__(self, defaults=None, disable_input=False, **kwargs):
                     css_class="disabled"
                 )
             )
+
+
+class ManageAppReleasesForm(forms.Form):
+    app_id = forms.ChoiceField(label=ugettext_lazy("Application"), choices=(), required=False)
+    location_id = forms.ChoiceField(label=ugettext_lazy("Location"), choices=(), required=False)
+    version = forms.IntegerField(label=ugettext_lazy('Version'), required=False)
+
+    def __init__(self, request, domain, *args, **kwargs):
+        self.domain = domain
+        super(ManageAppReleasesForm, self).__init__(*args, **kwargs)
+        self.fields['app_id'].choices = self.app_id_choices()
+        if request.GET.get('app_id'):
+            self.fields['app_id'].initial = request.GET.get('app_id')
+        self.fields['location_id'].choices = self.location_id_choices()
+        if request.GET.get('location_id'):
+            self.fields['location_id'].initial = request.GET.get('location_id')
+        if request.GET.get('version'):
+            self.fields['version'].initial = request.GET.get('version')
+        self.helper = FormHelper()
+        self.helper.label_class = 'col-sm-3 col-md-2'
+        self.helper.field_class = 'col-sm-9 col-md-8 col-lg-6'
+        self.helper.form_tag = False
+
+        self.helper.layout = crispy.Layout(
+            crispy.Field('app_id', id='app-id-search-select'),
```

Comment:
This should also be `ko-select2`, so all the selects on the page match. In general plain select boxes aren't great - long lists should use select2s and short lists should use button toggles.

Reply:
button toggles ?

Reply:
these toggles: https://www.commcarehq.org/styleguide/molecules/#molecules-selections

Reply:
aha! okay. I expect this to be long enough to keep it as select.

Reply:
agreed